### PR TITLE
Refactor package layout

### PR DESCRIPTION
**Pull request by nrc**
And add a few docs

PTAL @Connor1996

### CODE REVIEW

File: kv/tikv/server.go [hunk 1]
```diff
@@ -1,140 +1,85 @@
...
 func (server *Server) KvResolveLock(_ context.Context, req *kvrpcpb.ResolveLockRequest) (*kvrpcpb.ResolveLockResponse, error) {
-	cmd := NewResolveLock(req)
+	cmd := commands.NewResolveLock(req)
 	resp, err := server.Run(&cmd)
 	return resp.(*kvrpcpb.ResolveLockResponse), err
 }
 
-// Raw API.
+// Raw API. These commands are handled inline rather than by using Run and am implementation of the Commands interface.
+// This is because these commands are fairly straightforward and do not share a lot of code with the transactional
```
Comment (Connor1996) [inline kv/tikv/server.go]: it should share nothing related code with transactional commands.

File: kv/tikv/transaction/commands/command.go [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+		return resp, nil
+	}
+
+	return nil, err
+}
+
+// RawRegionError is similar to regionError but is specialized for the raw commands. They all have an `Error` field, so
+// if the error is not a region error, then it is set to that field in the response. This means that the error is always
+// handled and resp should be returned back to the gRPC layer.
+func RawRegionError(err error, resp interface{}) {
```
Comment (Connor1996) [inline kv/tikv/transaction/commands/command.go]: let's move it to `server.go`, raw API should have nothing dependency on the transaction package.

File: kv/tikv/server.go (hunk 1 above)
Comment (nrc) [reply kv/tikv/server.go]: To clarify, I meant that there is very little code that is the same in both. An example of shared code is that they both use dbreader, but through different abstractions.

File: kv/tikv/transaction/commands/command.go (hunk 2 above)
Comment (nrc) [reply kv/tikv/transaction/commands/command.go]: :+1:

File: kv/tikv/server.go (hunk 1 above)
Comment (Connor1996) [reply kv/tikv/server.go]: okay

**Comment by Connor1996** [comment]
Can we fix the problem issued in https://github.com/pingcap-incubator/tinykv/pull/78 here, or another PR?

**Comment by nrc** [comment]
Comments here and from #78 are addressed. @Connor1996 please take another look

File: kv/tikv/transaction/commands/command.go [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+	"reflect"
+
+	"github.com/pingcap-incubator/tinykv/kv/tikv/inner_server"
+	"github.com/pingcap-incubator/tinykv/kv/tikv/transaction/latches"
+	"github.com/pingcap-incubator/tinykv/kv/tikv/transaction/mvcc"
+	"github.com/pingcap-incubator/tinykv/proto/pkg/kvrpcpb"
+)
+
+// Command is an abstraction which covers the process from receiving a request from gRPC to returning a response.
+// That process is driven by a Scheduler.
```
Comment (Connor1996) [inline kv/tikv/transaction/commands/command.go]: should update

File: kv/tikv/transaction/doc.go [hunk 4]
```diff
@@ -19,5 +19,5 @@ package storage
 // underlying storage (or equivalently, you can think of every key having its own latch). TODO explain latching in more detail.
 //
 // Within this package, `commands` contains code to lower TinySQL requests to mvcc transactions. `exec` contains
```
Comment (Connor1996) [inline kv/tikv/transaction/doc.go]: should remove `exec` related description.

**Pull request merged**